pkg/k8shandler/forwarding.go

```diff
@@ -75,197 +46,196 @@ func (clusterRequest *ClusterLoggingRequest) generateCollectorConfig() (config s
-	if cluster.Spec.LogStore != nil && cluster.Spec.LogStore.Type == logging.LogStoreTypeElasticsearch {
-		if !clusterRequest.ForwardingSpec.DisableDefaultForwarding && len(clusterRequest.ForwardingSpec.Pipelines) == 0 {
-			logger.Debug("Configuring logforwarding to utilize the operator managed logstore")
-			if clusterRequest.ForwardingRequest.Status == nil {
-				clusterRequest.ForwardingRequest.Status = logforward.NewForwardingStatus(logforward.LogForwardingStateAccepted, logforward.LogForwardingReasonName, "")
-			}
-			clusterRequest.ForwardingRequest.Status.LogSources = []logforward.LogSourceType{logforward.LogSourceTypeApp, logforward.LogSourceTypeInfra}
-			clusterRequest.ForwardingRequest.Status.Outputs = []logforward.OutputStatus{
-				logforward.NewOutputStatus(
-					internalOutputName,
-					logforward.OutputStateAccepted,
-					logforward.OutputStateReasonConditionsMet,
-					"This is an operator generated output because forwarding is undefined and 'DisableDefaultForwarding' is false",
-				),
-			}
-			clusterRequest.ForwardingRequest.Status.Pipelines = []logforward.PipelineStatus{
-				logforward.NewPipelineStatus(
-					defaultAppPipelineName,
-					logforward.PipelineStateAccepted,
-					logforward.PipelineStateReasonConditionsMet,
-					"This is an operator generated pipeline because forwarding is undefined and 'DisableDefaultForwarding' is false",
-				),
-			}
-			return logforward.ForwardingSpec{
-				Outputs: []logforward.OutputSpec{
-					{
-						Name:     internalOutputName,
-						Type:     logforward.OutputTypeElasticsearch,
-						Endpoint: constants.LogStoreService,
-						Secret: &logforward.OutputSecretSpec{
-							Name: collectorSecretName,
-						},
-					},
-				},
-				Pipelines: []logforward.PipelineSpec{
-					{
-						Name:       defaultAppPipelineName,
-						SourceType: logforward.LogSourceTypeApp,
-						OutputRefs: []string{internalOutputName},
-					},
-					{
-						Name:       defaultInfraPipelineName,
-						SourceType: logforward.LogSourceTypeInfra,
-						OutputRefs: []string{internalOutputName},
-					},
-				},
-			}
+// normalizeForwarder normalizes the clusterRequest.ForwarderSpec, and
+// returns a normalized spec and status
+func (clusterRequest *ClusterLoggingRequest) normalizeForwarder() (*logging.ClusterLogForwarderSpec, *logging.ClusterLogForwarderStatus) {
+	logger.Debugf("Normalizing logforwarding from request: %v", clusterRequest)
+	logger.Debugf("ForwarderRequest: %v", clusterRequest.ForwarderRequest)
+	logger.Debugf("ForwarderSpec: %v", clusterRequest.ForwarderSpec)
+
+	// Default configuration for empty/missing forwarder, forward to the default store
+	if len(clusterRequest.ForwarderSpec.Pipelines) == 0 {
+		logger.Debug("Configuring forwarder to use the default log store")
+		clusterRequest.ForwarderSpec.Pipelines = []logging.PipelineSpec{
+			{
+				InputRefs:  []string{logging.InputNameApplication, logging.InputNameInfrastructure},
```

Comment:
Is there a specific reason we skip the audit logs for the default pipeline.

Reply:
Dunno, just following the TP forwarder.

Reply:
I'm following the original logforwarding impl here, not sure what the reason is.